### PR TITLE
Revert "Revert slack org does not exist changes breaking escalate command (#2057)"

### PR DESCRIPTION
**Pull request by joeyorlando**
# What this PR does

Closes https://github.com/grafana/oncall-private/issues/1836

- Revert "Revert slack org does not exist changes breaking escalate command (#2057)" + add some unit tests to ensure we don't break the `/escalate` command in the future
- cleanup how we destructure the `payload` dict in the endpoint handler

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/apps/slack/views.py [hunk 1]
```diff
@@ -285,55 +301,73 @@ def post(self, request):
                 # Open pop-up to inform user why OnCall bot doesn't work if any action was triggered
                 self._open_warning_window_if_needed(payload, slack_team_identity, warning_text)
                 return Response(status=200)
+        elif organization is None and payload_type_is_block_actions:
```
Comment (joeyorlando) [inline engine/apps/slack/views.py:304]: `and payload_type_is_block_actions` is the main change of this PR. 

From what I can see from the logs,`Organization matching query does not exist.` exceptions originating from this endpoint only seem to be happening when `payload_type == 'block_actions'`

File: engine/apps/slack/views.py [hunk 2]
```diff
@@ -161,8 +162,29 @@ def post(self, request):
...
+        payload_event_user = payload_event.get("user")
+        payload_event_bot_id = payload_event.get("bot_id")
+        payload_event_channel_type = payload_event.get("channel_type")
+
+        payload_event_message = payload_event.get("message", {})
+        payload_event_message_user = payload_event_message.get("user")
+
+        payload_event_previous_message = payload_event.get("previous_message", {})
+        payload_event_previous_message_user = payload_event_previous_message.get("user")
+
```
Comment (joeyorlando) [inline engine/apps/slack/views.py:185]: decided to clean this up a bit, much less repetitive imo

File: engine/apps/slack/tests/test_interactive_api_endpoint.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            "text": "potato",
+            "api_app_id": "A0909234092340293402934234234234234234",
+            "is_enterprise_install": "false",
+            "response_url": "https://hooks.slack.com/commands/cvcv/cvcv/cvcv",
+            "trigger_id": "asdfasdf.4122782784722.cvcv",
+        }
+    )
+
+    assert response.status_code == status.HTTP_200_OK
+    mock_open_warning_window_if_needed.assert_not_called()
```
Comment (joeyorlando) [inline engine/apps/slack/tests/test_interactive_api_endpoint.py:134]: previously this PR broke the `/escalate` command, added a test to ensure this is no longer the case. The request payload structure being sent here is identical to that when the `/escalate` command is used.

**Comment by joeyorlando** [comment]
closes https://github.com/grafana/oncall-private/issues/1836

**Review by matiasb — APPROVED** [review]
LGTM

**Pull request merged**